File: speedtest/cpp/NTupleTest.cpp
#include "TBranch.h"
#include "TChain.h"
#include "TFile.h"
#include "TH1.h"

#include "Read_FlatNtuple.h" // List of input branches and functions to return values

void NTupleTest(const char* inputFile, int maxEvents, const char* outFile)
{

  TFile* out = new TFile(outFile, "RECREATE");
  TH1D* lctHist = new TH1D("LCTTheta", "LCT Theta", 100, -50, 50);
  TH1D* trkHist = new TH1D("TrackPt", "Track pT", 100, 0, 100);

  TChain* in_chain = new TChain("FlatNtupleData/tree");
  in_chain->Add(inputFile);

  InitializeMaps();
  SetBranchAddresses(in_chain);

  int nEvents = in_chain->GetEntries();

  for (int iEvt = 0; iEvt < nEvents; iEvt++) {
    if (iEvt > maxEvents)
      break;
    if ((iEvt % 10000) == 0) {
      std::cout << "Looking at event " << iEvt << " out of " << nEvents
                << std::endl;
    }

    in_chain->GetEntry(iEvt);

    for (int i = 0; i < I("nHits"); i++) {
      if (I("hit_isCSC", i) != 1)
        continue;
      lctHist->Fill(F("hit_theta", i));
    }

    for (int trk = 0; trk < I("nTracks"); trk++) {
      trkHist->Fill(F("trk_pt", trk));
    }
  }

  delete in_chain;
  out->Write();
}

File: speedtest/python/NTupleTest.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import sys
import ROOT


def run(in_file, max_events, out_file):

    # Set ROOT options --------------------------------------------------------

    ROOT.gROOT.SetBatch(ROOT.kTRUE)

    # Prepare input/output ----------------------------------------------------

    # Create output file and load empty histograms
    output_file = ROOT.TFile(out_file, "RECREATE")
    lctHist = ROOT.TH1D("LCTTheta", "LCT Theta", 100, -50, 50)
    trkHist = ROOT.TH1D("TrackPt", "Track pT", 100, 0, 100)

    # Load ntuple files
    chain = ROOT.TChain("FlatNtupleData/tree")
    chain.Add(in_file)

    # Event loop --------------------------------------------------------------

    # Technically event is the same TChain pointer as chain, but iterating
    # updates the event index. Enumerate just lets counter hold the number of
    # events processed
    for counter, event in enumerate(chain):
        if counter % 10000 == 0:
            print("Processed {0} events".format(counter))
        if counter > max_events:
            break

        for lct in range(event.nHits):
            if not event.hit_isCSC[lct]:
                continue
            lctHist.Fill(event.hit_theta[lct])

        for trk in range(event.nTracks):
            trkHist.Fill(event.trk_pt[trk])


if __name__ == "__main__":
    run(sys.argv[1], int(sys.argv[2]), sys.argv[3])
